### PR TITLE
Rename the vault to the originsrv

### PR DESCRIPTION
**Pull request by adamhjk**
It became clear in the process of refactoring that the vault was
ill-named - in truth, it's the `originsrv` - the service responsible for
supplying data that's related to a given origin. This PR renames the
service.

Additionally, it makes the `make bldr-run` command work again. It does
this by firing up a postgresql instance within hab-sup, and creating the
default databases needed by the services.

A word of caution - this will re-configure your database instance.

Another word of caution - this will conflict with running the
integration tests for components with a database.

gif-keyboard-7341132734829651417

Signed-off-by: Adam Jacob <adam@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-db/src/config.rs [hunk 1]
```diff
@@ -15,5 +15,5 @@
 use num_cpus;
 
 pub fn default_pool_size() -> u32 {
-    (num_cpus::get() * 8) as u32
+    (num_cpus::get() * 2) as u32
```
Comment (reset) [inline components/builder-db/src/config.rs]: Did you intend to drop this down to 2 from 8 in the rename?

**Comment by reset** [comment]
@adamhjk yep, this is what the appropriate name is now that we've got a more complete picture of the build service.

One minor comment inline but the rest looks good. Merge when ready!

gif-keyboard-6954068068502647158

File: components/builder-db/src/config.rs (hunk 1 above)
Comment (adamhjk) [inline components/builder-db/src/config.rs]: I did! During the course of digging in to PostgreSQL literature, the formula they recommended was `(CPU * 2) + NUM_SPINDLES`. 

The CPU*8 was blowing the default connection limits in postgresql. We could certainly up them (and you know we would in production), but even then I suspect this is the rightish number.

**Comment by reset** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/209560572).

File: components/builder-db/src/config.rs (hunk 1 above)
Comment (reset) [inline components/builder-db/src/config.rs]: Makes perfect sense, just wanted to make sure this was an intended change!

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/209560572).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**